nixos/modules/services/security/vaultwarden/default.nix

```
@@ -194,18 +202,21 @@ in {
         ProtectHome = "true";
         ProtectSystem = "strict";
         AmbientCapabilities = "CAP_NET_BIND_SERVICE";
-        StateDirectory = "bitwarden_rs";
-        StateDirectoryMode = "0700";
         Restart = "always";
-      };
+      } // (if lib.hasPrefix "/var/lib/" cfg.stateDir then {
+        StateDirectory = builtins.baseNameOf cfg.stateDir;
```

```suggestion
        StateDirectory = lib.removePrefix "/var/lib/" cfg.stateDir;
```
because otherwise e.g. `/var/lib/foo/bar` wouldn't work